VideoWeb/VideoWeb/Controllers/MagicLinksController.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            _videoApiClient = videoApiClient;
+            _logger = logger;
+        }
+
+        [HttpGet("GetMagicLinkParticipantRoles")]
+        [AllowAnonymous]
+        [SwaggerOperation(OperationId = "GetMagicLinkParticipantRoles")]
+        [ProducesResponseType(typeof(List<Role>), StatusCodes.Status200OK)]
+        public IActionResult GetMagicLinkParticipantRoles()
+        {
+            var magicParticipantRoles = new List<Role>
+            {
+                Role.MagicLinkParticipant, Role.MagicLinkObserver
+            };
+
+            return Ok(magicParticipantRoles);
+        }
+
+        [HttpGet("ValidateMagicLink/{hearingId}")]
+        [AllowAnonymous]
+        [SwaggerOperation(OperationId = "ValidateMagicLink")]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        public async Task<IActionResult> ValidateMagicLink(Guid hearingId)
+        {
+            try
+            {
+                var response = await _videoApiClient.ValidateMagicLinkAsync(hearingId);
+                return Ok(response);
+            }
+            catch(VideoApiException e)
+            {
+                _logger.LogError(e, $"Unable to get conference with hearing id: {hearingId}");
+                return StatusCode(e.StatusCode, e.Response);
+            }
+        }
+
+        [HttpPost("joinConferenceAsAMagicLinkUser/${hearingId}")]
+        [AllowAnonymous]
+        [SwaggerOperation("joinConferenceAsAMagicLinkUser")]
+        [ProducesResponseType(typeof(MagicLinkParticipantJoinResponse), (int) HttpStatusCode.OK)]
+        public async Task<IActionResult> Join(Guid hearingId,
+            [FromBody] MagicLinkParticipantJoinRequest joinRequest)
+        {
+            var roleAsUserRole = (UserRole)Enum.Parse(typeof(UserRole), joinRequest.Role.ToString());
+
+            if (roleAsUserRole != UserRole.MagicLinkObserver && roleAsUserRole != UserRole.MagicLinkParticipant)
+            {
+                throw new NotSupportedException(
+                    $"Can only join as a magic user if the roles are MagicLinkParticipant or MagicLinkObserver. The Role was {roleAsUserRole}");
+            }
+            
+            MagicLinkParticipantJoinResponse joinResponse = new MagicLinkParticipantJoinResponse()
+            {
+                Jwt = await _videoApiClient.AddMagicLinkParticipantAsync(hearingId, new AddMagicLinkParticipantRequest()
+                {
+                    Name = joinRequest.Name,
+                    UserRole = roleAsUserRole
+                })
+            };
+            
+            return Ok(joinResponse);
+        }
+        
+        [HttpGet("isMagicLinkParticipantAuthorised")]
+        [SwaggerOperation("joinConferenceAsAMagicLinkUser")]
+        [ProducesResponseType((int) HttpStatusCode.OK)]
+        [ProducesResponseType((int) HttpStatusCode.Forbidden)]
+        public IActionResult IsAuthorised()
```

Comment:
Where is this used?

Reply:
Its called from the frontend, in the checkAuth method of the magicLinksSecurityService. It is just an endpoint that will return 401 if the JWT is invalid or a 200 if it is valid.